test/constant.cpp

```diff
@@ -49,7 +49,7 @@ int main()
         static_assert(
             std::is_same<
                 decltype(1844674407370955161_c),
-                smath::constant<long long, 1844674407370955161>
+                smath::constant<long, 1844674407370955161>
```

Comment:
I don't know where else to ask, but how does this build on windows? I tried just doing a `cmake -G"Visual Studio 14 2015 Win64"` but it fails to build with a large amount of errors.